Add docs for depend-defaults (#119)
microsoft/vcpkg-docs: vcpkg/reference/vcpkg-json.md

See [versioning](../users/versioning.md#baselines) and [Using registries](../users/registries.md) for more semantic details.

### <a name="depend-defaults"></a> `"depend-defaults"`

The default value for the [`"default-features"`](#dependency-default-features) field of dependencies. A boolean. `true` by default. Optional.

> If used at a top-level manifest, this field diables default features for all ports in the dependency tree. If used in a port, this field sets the default value for all dependencies in the port.

Reviewer: ```suggestion
If used in a top-level manifest, this field disables [default features](#default-features) for all ports in the dependency tree. If used in a port, this field disables the default features for the port's dependencies only.
```

Or perhaps it would be better to rearrange it?

> Disables [default features](#default-features) for the port's dependencies. If used in the top-level manifest, disables default features for the whole dependency tree.

Author: The rearrange version is shorter which is better. I think the main motivation to mention it the other way around was that for end users the "top level thing" is more interesting.
